plugins/linux/ifplugin/interface_state.go

```diff
@@ -75,12 +75,21 @@ func (c *LinuxInterfaceStateUpdater) Init(ctx context.Context, logger logging.Pl
 
 // Close watcher channel (state chan is closed in LinuxInterfaceConfigurator)
 func (c *LinuxInterfaceStateUpdater) Close() error {
-	if err := safeclose.Close(c.ifWatcherNotifCh, c.ifWatcherDoneCh); err != nil {
+	if err := safeclose.Close(c.ifWatcherNotifCh, c.ifWatcherDoneCh, c.ifStateChan); err != nil {
```

Comment:
You should not close `ifStateChan` here, as it runs in different goroutine than the producer that puts data into this channel, gouroutine that runs `watchLinuxInterfaces`.